### PR TITLE
Added Recent Screen Test

### PR DESCRIPTION
**Pull request by zachwhite99**
Added test to verify that pressed buttons would show up in "Recents" category
---

### CODE REVIEW

File: VocableUITests/Screens/MainScreen.swift [hunk 1]
```diff
@@ -28,13 +28,19 @@ class MainScreen {
     func isTextDisplayed(_ text: String) -> Bool {
         return app.collectionViews.staticTexts[text].exists
     }
-    
-    func scrollRightAndTapCurrentCategory(numTimesToScroll: Int) {
+       
+    // added a parameter currentCategory, so that it is not always assumed started at "General"
+    func scrollRightAndTapCurrentCategory(numTimesToScroll: Int, currentCategory: String) {
```
Comment (stechkev) [inline VocableUITests/Screens/MainScreen.swift]: we refer to current category twice here but they reference different things. we should update either the name of the method or the variable to better capture the behavior

File: VocableUITests/Screens/MainScreen.swift [hunk 2]
```diff
@@ -28,13 +28,19 @@ class MainScreen {
...
+        
         for _ in 1...numTimesToScroll {
             categoryRightButton.tap()
         }
-        let currentCategory = numTimesToScroll % defaultCategories.count
-        app.collectionViews.staticTexts[defaultCategories[currentCategory]].tap()
+        
+        // let currentCategory = numTimesToScroll % defaultCategories.count
```
Comment (stechkev) [inline VocableUITests/Screens/MainScreen.swift]: should delete unused code unless it was left for a particular reason

File: VocableUITests/Screens/MainScreen.swift [hunk 3]
```diff
@@ -28,13 +28,19 @@ class MainScreen {
...
+        let currentPosition = defaultCategories.firstIndex(of: currentCategory)!
+        // FIXED: changed +1 to +numTimesToScroll
```
Comment (stechkev) [inline VocableUITests/Screens/MainScreen.swift]: same with this comment

File: VocableUITests/Tests/MainScreenTests.swift [hunk 4]
```diff
@@ -10,6 +10,32 @@ import XCTest
 
 class MainScreenTests: BaseTest {
     
+    func testRecentScreen(){
+            /*
+         for each categoryName (the first 5 categories), tap() the top left
+         speech button, then verify that all pressed buttons appear in "Recents"
+            */
```
Comment (stechkev) [inline VocableUITests/Tests/MainScreenTests.swift]: i would move this to right above the test name

File: VocableUITests/Tests/MainScreenTests.swift [hunk 5]
```diff
@@ -10,6 +10,32 @@ import XCTest
...
+        let app = XCUIApplication()
+        
+        var listOfPressedButtons: [String] = []
+        var buttonText = ""
+        
+        for categoryName in mainScreen.defaultCategories {
+            if categoryName == "123" { // "123" categoryName does not add to RECENTS category
+                break;
+            }
+            buttonText = app.collectionViews.staticTexts.element(boundBy: 0).label // first button's label
```
Comment (stechkev) [inline VocableUITests/Tests/MainScreenTests.swift]: get rid of comment and rename variable to firstButtonText

File: VocableUITests/Tests/MainScreenTests.swift [hunk 6]
```diff
@@ -10,6 +10,32 @@ import XCTest
...
+            app.collectionViews.staticTexts[buttonText].tap() // tap top left button
+            listOfPressedButtons.append(buttonText) // add that button's text to listOfPressedButtons
```
Comment (stechkev) [inline VocableUITests/Tests/MainScreenTests.swift]: probs don't need the comments here either

File: VocableUITests/Tests/MainScreenTests.swift [hunk 7]
```diff
@@ -10,6 +10,32 @@ import XCTest
...
+            mainScreen.scrollRightAndTapCurrentCategory(numTimesToScroll: 1, currentCategory: categoryName)
+        }
+        mainScreen.scrollRightAndTapCurrentCategory(numTimesToScroll: 2, currentCategory: "123")
+        
+        for text in listOfPressedButtons {
```
Comment (stechkev) [inline VocableUITests/Tests/MainScreenTests.swift]: text can be something a bit more descriptive

**Review by stechkev — CHANGES_REQUESTED** [review]
found a couple things to change but overall logic looks really good!

File: VocableUITests/Screens/MainScreen.swift (hunk 1 above)
Comment (zachwhite99) [reply VocableUITests/Screens/MainScreen.swift]: changed parameter name to "startingCategory", edited all callers that use this function to reflect that change

File: VocableUITests/Screens/MainScreen.swift (hunk 2 above)
Comment (zachwhite99) [reply VocableUITests/Screens/MainScreen.swift]: deleted!

File: VocableUITests/Screens/MainScreen.swift (hunk 3 above)
Comment (zachwhite99) [reply VocableUITests/Screens/MainScreen.swift]: deleted!

File: VocableUITests/Tests/MainScreenTests.swift (hunk 4 above)
Comment (zachwhite99) [reply VocableUITests/Tests/MainScreenTests.swift]: moved to above test name

File: VocableUITests/Tests/MainScreenTests.swift (hunk 5 above)
Comment (zachwhite99) [reply VocableUITests/Tests/MainScreenTests.swift]: renamed variable to "firstButtonText"

File: VocableUITests/Tests/MainScreenTests.swift (hunk 6 above)
Comment (zachwhite99) [reply VocableUITests/Tests/MainScreenTests.swift]: removed comments here

File: VocableUITests/Tests/MainScreenTests.swift (hunk 7 above)
Comment (zachwhite99) [reply VocableUITests/Tests/MainScreenTests.swift]: changed "text" variable name to "pressedButtonText"

**Review by stechkev — APPROVED** [review]
looks good!

**Pull request merged**